ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -464,6 +476,26 @@
     return result;
   }
 
+  private Optional<UInt64> computeEarliestBlobSidecarsSlot(
+      final BlobSidecarsAndValidationResult blobSidecarsAndValidationResult,
+      final BeaconBlock block) {
+    if (!blobSidecarsAndValidationResult.isValid()) {
+      return Optional.empty();
+    }
+    // Earliest empty slot before this block or slot of this block if no empty (without blocks)
+    // slots before. If it's inside Deneb, it's `earliestBlobSidecarsSlot`
```

Comment:
This comment is driving me crazy. We need to describe why we need this code instead of writing the code in english :)

something like that, as comment for the entire method

\\ In order to keep track of DataAvailability Window, we need to compute the earliest slot we can consider data available for the given block. It needs to take in account possible empty slots so the actual slot should be parent's block slot + 1. Moreover, we need to manage the case in which the calculated slot is before Deneb activation, so it should never be before the first deneb slot.